e2e/cypress/integration/messaging/pinned_posts_part2_spec.js

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @messaging
+
+import * as TIMEOUTS from '../../fixtures/timeouts';
+import {getAdminAccount} from '../../support/env';
+
+describe('Pinned posts part 2', () => {
+    let testTeam;
+    let testChannel;
+    let testUser;
+    let otherUser;
+
+    before(() => {
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            testUser = user;
+            testTeam = team;
+            testChannel = channel;
+            return cy.apiCreateUser();
+        }).then(({user: user1}) => {
+            otherUser = user1;
+            return cy.apiAddUserToTeam(testTeam.id, otherUser.id);
+        }).then(() => {
+            cy.apiAddUserToChannel(testChannel.id, otherUser.id);
+        });
+    });
+
+    it('MM-T2173 Un-pinning a post from reply RHS also removes badge in center', () => {
+        // # Login
+        cy.apiLogin(testUser);
+
+        // # Visit a test channel and post a message
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+        cy.postMessage('Hello');
+
+        cy.getLastPostId().then((postId) => {
+            // # Pin the post
+            pinPost(postId);
+
+            // # Now open RHS
+            cy.get(`#post_${postId}`).trigger('mouseover');
+            cy.get(`#CENTER_commentIcon_${postId}`).click({force: true});
+
+            // * Verify that pinned icon is there both in center and RHS
+            cy.get(`#rhsPost_${postId} .post-pre-header`).should('have.text', 'Pinned');
+            cy.get(`#post_${postId} .post-pre-header`).should('have.text', 'Pinned');
+
+            // # Un-pin the post from RHS
+            unpinPost(postId);
+
+            // * Verify that pinned icon is gone from center and RHS
+            cy.get(`#rhsPost_${postId} .post-pre-header`).should('not.have.text', 'Pinned');
+            cy.get(`#post_${postId} .post-pre-header`).should('not.have.text', 'Pinned');
+
+            // # Click pin icon
+            cy.get('#channelHeaderPinButton').should('be.visible').click();
+
+            // * Should not have any pinned posts
+            cy.get('#search-items-container .no-results__title').should('have.text', 'No pinned posts yet');
+        });
+    });
+
+    it('MM-T2174 Switching channels in center also changes pinned posts RHS', () => {
+        // # Login
+        cy.apiLogin(testUser);
+
+        // # Visit a test channel
```

Comment:
Hi @agnivade - do you need to add `and post a message` to this line and to the code below?

Reply:
The post already exists in the channel. So we don't want to add a new one. Getting the last post is sufficient.

Reply:
Ah, ok... For every other test in the PR you have code to make a post first and it was missing on this one so I just wondered.  Also, will these tests run concurrently in Cypress or does each test run individually from start to finish? (Just trying to understand how we re-use a post from a previous test case why the tests are running)

Reply:
They run individually from start to finish. It's exactly like how you would manually run the tests by hand one by one.